feature/experimental/system/gnmi/benchmarking/ate_tests/internal/setup/setup.go

```diff
@@ -0,0 +1,509 @@
+// RouteCount and IsisInstance are used in other files which import
+// setup.go
+const (
+	// DutAS can be exported
+	DutAS = 64500
+	// AteAs can be exported
+	AteAS  = 64501
+	ateAS2 = 64502
+	// PeerGrpName can be exported
+	PeerGrpName    = "BGP-PEER-GROUP"
+	plenIPv4       = 30
+	dutStartIPAddr = "192.0.2.1"
+	ateStartIPAddr = "192.0.2.2"
+	// RouteCount can be exported
+	RouteCount            = 200
+	advertiseBGPRoutesv4  = "203.0.113.1"
+	authPassword          = "ISISAuthPassword"
+	advertiseISISRoutesv4 = "198.18.0.0"
+	IsisInstance          = "DEFAULT"
+)
+
+var (
+	// DutIPPool can be exported
+	DutIPPool = make(map[string]net.IP)
+	// AteIPPool can be exported
+	AteIPPool = make(map[string]net.IP)
+)
+
+// M is type variable and can be exported
+type M map[string]interface{}
+
+// BuildIPPool is to Build pool of ip addresses for both DUT and ATE interfaces.
+// It reads ports given in binding file to calculate ip addresses needed.
+func BuildIPPool(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	var dutIPIndex, ipSubnet, ateIPIndex int = 1, 2, 2
+	var endSubnetIndex = 253
+	for _, dp := range dut.Ports() {
+		dutNextIP := nextIP(net.ParseIP(dutStartIPAddr), dutIPIndex, ipSubnet)
+		ateNextIP := nextIP(net.ParseIP(ateStartIPAddr), ateIPIndex, ipSubnet)
+		DutIPPool[dp.ID()] = dutNextIP
+		AteIPPool[dp.ID()] = ateNextIP
+
+		// Increment dut and ate host ip index by 4
+		dutIPIndex = dutIPIndex + 4
+		ateIPIndex = ateIPIndex + 4
+
+		// Reset dut and ate IP indexes when it is greater endSubnetIndex
+		if dutIPIndex > int(endSubnetIndex) {
+			ipSubnet = ipSubnet + 1
+			dutIPIndex = 1
+			ateIPIndex = 2
+		}
+	}
+
+}
+
+// nextIP returns ip address based on hostindex and subnetindex provided.
+func nextIP(ip net.IP, hostIndex int, subnetIndex int) net.IP {
+	s := ip.String()
+	sa := strings.Split(s, ".")
+	sa[2] = strconv.Itoa(subnetIndex)
+	sa[3] = strconv.Itoa(hostIndex)
+	s = strings.Join(sa, ".")
+	return net.ParseIP(s)
+}
+
+// CreateGNMIUpdate is to create GNMI update message . It will marshal the input
+// strings provided and return gpb update message to the calling function
+func CreateGNMIUpdate(map1 string, map2 string, configElem []M) *gpb.Update {
+
+	j := map[string]interface{}{
+		map1: map[string]interface{}{
+			map2: configElem,
+		},
+	}
+
+	v, err := json.Marshal(j)
+	if err != nil {
+		err1 := fmt.Errorf("marshal of intf config failed with unexpected error: %v", err)
+		fmt.Println(err1.Error())
+	}
+	update := &gpb.Update{
+		Path: &gpb.Path{Elem: []*gpb.PathElem{}},
+		Val:  &gpb.TypedValue{Value: &gpb.TypedValue_JsonIetfVal{JsonIetfVal: v}},
+	}
+	return update
+}
+
+// BuildOCInterfaceUpdate function is to build  OC config for interfaces.
+// It reads ports from binding file and returns gpb update message
+// which will have configurations for all the ports.
+func BuildOCInterfaceUpdate(t *testing.T) *gpb.Update {
+	dut := ondatra.DUT(t, "dut")
+	var intfConfig []M
+
+	for _, dp := range dut.Ports() {
```

Comment:
Throughout this code, please can we use ygot-generated structs as per other test cases. This kind of approach means that if the schema changes there is no compile time checking that would catch that this test should fail.

There are examples in other tests ([example](https://github.com/openconfig/featureprofiles/blob/main/feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go#L114)). 

For this case, this should look something like:


```golang
i := &telemetry.Interface{
	Enabled: ygot.Bool(true),
	Description: ygot.String("from oc"),
	Name: ygot.String(dp.Name()),
	Type: telemetry.IETFInterfaces_InterfaceType_ethernetCsmacd,
}

i.GetOrCreateSubinterface(0).GetOrCreateAddress(DUTIPPool[dp.ID()]).PrefixLength = ygot.Uint8(plenIPv4)
```

Which can then be used with `dut.Config().Interface(dp.Name()).Update(t, i)`.

Reply:
@robshakir , Thanks for reviewing.

gnmi1.3 test is a benchmarking testcase and as per test we need to apply the config in one shot , hence I have embeded json map struct , which will apply config for all the interfaces, protocols at once. 

If we use update/replace function, it will apply config only one interface at a time.

I can update code to use ygot data types like below. Please check if this looks good.

elem := map[string]interface{}{
			"name": ygot.String(dp.Name()),
			"config": map[string]interface{}{
				"enabled":     ygot.Bool(true),
				"description": ygot.String("from oc"),
				"name":        ygot.String(dp.Name()),
				"type":        telemetry.IETFInterfaces_InterfaceType_ethernetCsmacd,
			},

Thanks,
Prabha

Reply:
We can generate the different structs that we need and then use `ygot.MergeStructs` or `ygot.MergeStructsInto` to create a larger configuration. This configuration can then be used in the update or replace.

The proposal here doesn't really use "ygot" types, these are just methods that use pointers.

IMHO, we should not have any "shadow" definition of the schema that uses `map[string]interface{}`. There are mature tools that are being used for OC generation throughout the featureprofiles repo, and don't suffer from the issue of not being tied to the underlying schema such that they will not be detected as failure until runtime against a device (when they become harder to distinguish as test programming issues).

Reply:
@robshakir , please could you point me to some examples where we can create larger configurations.
Example which i came across are mostly using update/replace per interface basis.

Reply:
Here is an example for [aggregate interfaces](https://github.com/openconfig/featureprofiles/blob/b8054dcbfa3c998035fd2dc8940991dd314553fc/feature/interface/aggregate/aggregate_test.go).

Reply:
@dplore , Thanks Darren.
We will check this and update the script.

Thanks,
Prabha

Reply:
@robshakir , @dplore , I have updated with review comments, please could you review. 

Thanks,
Prabha